src/main.h

```diff
@@ -81,7 +81,7 @@ static const unsigned int MAX_TX_SIGOPS = MAX_BLOCK_SIGOPS/5;
 /** Default for -maxorphantx, maximum number of orphan transactions kept in memory */
 static const unsigned int DEFAULT_MAX_ORPHAN_TRANSACTIONS = 100;
 /** The maximum size of a blk?????.dat file (since 0.8) */
-static const unsigned int MAX_BLOCKFILE_SIZE = 0x8000000; // 128 MiB
+static const unsigned int MAX_BLOCKFILE_SIZE = 0x7f000000; // (2048 - 16) MiB
```

Comment:
I understand the need, but be aware this will have detrimental effects on e.g. master datadir feature (PR 13746 in bitcoin), as it copies the most recent block and rev file(s) over.

Reply:
I don't think it does..? The only thing that `MAX_BLOCKFILE_SIZE` controls is the decision about whether to allocate a new file or increase the size of the current file when it tries to allocate new space. The rest of the code AFAICT doesn't  care in the slightest what size the files are.

Reply:
The master datadir code basically copies over the latest block file, so instead of a few hundred MBs, you get GBs here. I don't think it's a big deal, though.

Reply:
Oh, ok. Yeah you'd end up with a bit more excess wasteful copying, up to 2GB in size. Not a deal breaker though.

Reply:
I've found a way to resolve this issue. By comparing against `GetAdjustedTime` we can check if we are running in the protocol cleanup regime, and only then use the higher limits for block file size. In the mean time we'll still use smaller block files which is beneficial for pruning.

It's slightly layer violating to access the chain consensus parameters in some cases, e.g. from the network and protocol message handling code. But it doesn't break the build and is better than the alternative.

I've already coded up a patch to do this.